add heading ID suggester script file and workflow (#3849)
SoftwareAG/c8y-docs: scripts/documentation-style-guide.md

```markdown

> # Cumulocity Documentation Style Guide - Copilot Instructions

Reviewer: ```suggestion
# Cumulocity documentation guidelines
```

> ## Structure & Organization

Reviewer: ```suggestion
## Structure & organization
```

### Headings
- Use sentence capitalization
- Structure: "To [action] [object]" for procedures
- Maximum 5 levels (use bold for deeper levels)

### Heading IDs
- If a heading has **no ID**, generate the correct ID.
- If a heading **has an ID**, validate it and suggest a correction if it does not match the rules.
- Create the ID using:
  - Use lowercase letters only
  - Use hyphens to separate words
  - Remove spaces, punctuation, and special characters
  - Match the heading text
  - Keep the ID stable to avoid broken links

Example:  
`### Tech community {#tech-community}`

> ## Terminology Standards

Reviewer: ```suggestion
## Terminology standards
```

Reviewer: As the rule says: Do not capitalize :-)

### Capitalization
When in doubt, don’t capitalize.

Use **sentence-style capitalization** as the default. This means:

> - Capitalize only the **first word** of a sentence, heading, title, UI label (buttons, checkboxes), or standalone phrase.

Reviewer: ```suggestion
- Capitalize only the **first word** of a heading, title, UI label (buttons, checkboxes), or a sentence in a text.
```

- Keep all other words **lowercase**, unless they are proper nouns.
- In programming and code snippets, follow the capitalization norms of the language.
- **Do not use ALL CAPS** for emphasis.

**Capitalization checklist**

> - Lowercase everything except the first word in a sentence, UI label, phrase, heading, or title.

Reviewer: ```suggestion
- Lowercase everything except the first word in a UI label, phrase, heading, title, or sentence in a text.
```

> - Capitalize proper nouns.

Reviewer: ```suggestion
- Capitalize proper nouns if spelled that way.
```
